src/appleseed/renderer/modeling/shadergroup/shadergroup.cpp

```diff
@@ -227,6 +229,7 @@ bool ShaderGroup::create_optimized_osl_shader_group(
         report_has_closure(g_holdout_str.c_str(), HasHoldout);
         report_has_closure(g_debug_str.c_str(), HasDebug);
 
+        report_has_closure(g_mattesurface_str.c_str(), HasMatte);
```

Comment:
Move after the next line to respect the order you've followed everywhere (matte after NPR).